### PR TITLE
Color sensor

### PR DESCRIPTION
**Pull request by mud-ali**

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+import com.qualcomm.robotcore.eventloop.opmode.LinearOpMode;
+import com.qualcomm.robotcore.hardware.ColorSensor;
+import com.qualcomm.robotcore.hardware.HardwareMap;
+
+@Config
+public class ColorDetect {
+    ColorSensor colorSensor;
+
+    public ColorDetect(HardwareMap hMap) {
+        this.colorSensor = hMap.get(ColorSensor.class, "color");
```
Comment (mud-ali) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:18]: probably should be adding this to config file

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    /**
+     *  @return a String "red", "white", or "blue" or something
+     */
+    public String readColor() {
+        float[] hsv = this.readHSV();
+        float hue = hsv[0];
+        String color = "";
+        if(hue<=127) {
+            color = "yellow";
```
Comment (mud-ali) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java]: probably should change "yellow, pink, blue" to "red, white, blue"

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+@Config
+public class ColorDetect {
+    ColorSensor colorSensor;
+
+    public ColorDetect(HardwareMap hMap) {
+        this.colorSensor = hMap.get(ColorSensor.class, "color");
+    }
+
+    public int[] readRGB() {
```
Comment (mud-ali) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:21]: no javadoc comment 😔

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java (hunk 2 above)
Comment (mud-ali) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java]: and make sure numbers are adjusted accordingly (idk how it works ill believe you)

**Review by mud-ali — COMMENTED** [review]
woo

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java (hunk 3 above)
Comment (devin-joe) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:21]: what is javadoc

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java (hunk 1 above)
Comment (devin-joe) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:18]: i do not know what that means

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java (hunk 3 above)
Comment (mud-ali) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:21]: > what is javadoc

the thing you do with like the @return @params and all that stuff @varghs might explain better

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java (hunk 1 above)
Comment (mud-ali) [reply TeamCode/src/main/java/org/firstinspires/ftc/teamcode/utilities/ColorDetect.java:18]: you need to add ports for all hardware objects in the file called CONFIG.java

**Comment by mud-ali** [comment]
if youre changing stuff, pull first. otherwise lmk and we can merge this

**Pull request merged**